cmd/filestransfer/filestransfer.go

```diff
@@ -0,0 +1,164 @@
+	protocol := env("SNOWFLAKE_TEST_PROTOCOL", false)
+
+	port := 443
+	var err error
+	if len(portStr) > 0 {
+		port, err = strconv.Atoi(portStr)
+		if err != nil {
+			return "", nil, err
+		}
+	}
+
+	cfg := &sf.Config{
+		Account:   account,
+		User:      user,
+		Password:  password,
+		Database:  database,
+		Warehouse: warehouse,
+		Schema:    schema,
+		Host:      host,
+		Port:      port,
+		Protocol:  protocol,
+	}
+
+	dsn, err := sf.DSN(cfg)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return dsn, cfg, nil
+}
+
+func decompressAndRead(file *os.File) (string, error) {
+	gzipReader, err := gzip.NewReader(file)
+	defer gzipReader.Close()
+	if err != nil {
+		return "", err
+	}
+	var b bytes.Buffer
+	_, err = b.ReadFrom(gzipReader)
+	if err != nil {
+		return "", err
+	}
+	return b.String(), nil
+}
+
+func printRows(rows *sql.Rows) {
+	i := 1
+	for rows.Next() {
+		var col1, col2 string
+		if err := rows.Scan(&col1, &col2); err != nil {
+			log.Fatalf("error while scaning rows; err: %v", err)
+		}
+		fmt.Printf("Row %v: %v, %v\n", i, col1, col2)
+		i++
+	}
+}
+
+func main() {
+	if !flag.Parsed() {
+		flag.Parse()
+	}
+
+	//Opening connection
+	dsn, cfg, err := getDSN()
+	if err != nil {
+		log.Fatalf("error while creating DSN from config: %v, error: %v", cfg, err)
+	}
+	db, err := sql.Open("snowflake", dsn)
+
+	//Creating table to which the data from CSV file will be copied
+	_, err = db.Exec("CREATE OR REPLACE TABLE TEST_FILE_DATA(num integer, text varchar);")
+	if err != nil {
+		log.Fatalf("error while creating table; err: %v", err)
+	}
+	defer db.Close()
+
+	//Uploading data_to_upload.csv to internal stage for table TEST_FILE_DATA
+	currentDir, _ := os.Getwd()
+	filePath := currentDir + "/cmd/filestransfer/data_to_upload.csv"
+	_, err = db.Exec("PUT file://" + filePath + " @%TEST_FILE_DATA;")
+	if err != nil {
+		log.Fatalf("error while uploading file; err: %v", err)
+	}
+	fmt.Println("data_do_upload.csv successfully uploaded to internal stage.")
+
+	//Creating custom file format that describes our data
+	_, err = db.Exec("CREATE OR REPLACE TEMPORARY FILE FORMAT CUSTOM_CSV_FORMAT" +
+		" TYPE = CSV COMPRESSION = GZIP FIELD_DELIMITER = ';' FILE_EXTENSION = 'csv' SKIP_HEADER = 1;")
+	if err != nil {
+		log.Fatalf("error while creating file format; err: %v", err)
+	}
+	fmt.Println("Custom CSV format successfully created.")
+
+	//Loading data from files in stage area into table
+	_, err = db.Exec("COPY INTO TEST_FILE_DATA FILE_FORMAT = CUSTOM_CSV_FORMAT;")
+	if err != nil {
+		log.Fatalf("error while copying data into table; err: %v", err)
+	}
+	fmt.Println("Data successfully loaded into table.")
+
+	//Querying loaded data from table
+	rows, err := db.Query("SELECT * FROM TEST_FILE_DATA;")
+	defer rows.Close()
+	if err != nil {
+		log.Fatalf("error while querying data from table; err: %v", err)
+	}
+	printRows(rows)
+
+	//Downloading file from stage area
+	_, err = db.Exec("GET @%TEST_FILE_DATA/data_to_upload.csv file:///tmp/;")
```

Comment:
I suggest https://yourbasic.org/golang/temporary-file-directory/ as different OS may have different temp directory.

Reply:
Used `os.TempDir()` as ioutil method is deprecated.